src/components/builds/ConfigurationWithIssues.tsx

```diff
@@ -89,28 +97,86 @@ function ConfigurationWithIssues(props: Props) {
     );
   }
 
-  const lines = build.parsingResult.processedYamlConfig.split('\n').map((lineContent, zeroBasedLineNumber) => [
-    <tr>
-      <td className={classes.lineNumber}>{zeroBasedLineNumber + 1}</td>
-      <td className={classes.lineContent}>
-        <span>{lineContent}</span>
-      </td>
-    </tr>,
-    getIssuesForLine(zeroBasedLineNumber + 1),
-  ]);
+  function generateTable(configuration, issueCache: Map<number, any>): ReactNode {
+    if (issueCache.size === 0) return null;
+
+    const tableContents = configuration.split('\n').map((lineContent, zeroBasedLineNumber) => [
+      <tr>
+        <td className={classes.lineNumber}>{zeroBasedLineNumber + 1}</td>
+        <td className={classes.lineContent}>
+          <span>{lineContent}</span>
+        </td>
+      </tr>,
+      getIssuesForLine(issueCache, zeroBasedLineNumber + 1),
+    ]);
+
+    const extraLines = issueCache.get(0)
+      ? issueCache.get(0).map(issue => [
+          <Alert className={classes.issue} severity={issue.level.toLowerCase()}>
+            {issue.message}
+          </Alert>,
+        ])
+      : null;
+
+    return (
+      <>
+        <table className={classes.configurationTable} cellPadding={0}>
+          <tbody>{tableContents}</tbody>
+        </table>
+        {extraLines != null && <br />}
+        {extraLines}
+      </>
+    );
+  }
 
   const errorIssue = build.parsingResult.issues.find(it => it.level === 'ERROR');
-  const summaryText = errorIssue ? 'Failed to parse configuration' : 'Issues found while parsing configuration';
+
+  const yamlIssueCache = cacheIssues(
+    build.parsingResult.issues.filter(
+      issue => issue.path.endsWith('.cirrus.yml') || issue.path.endsWith('.cirrus.yaml'),
+    ),
+  );
+  const yamlTable = generateTable(build.parsingResult.processedYamlConfig, yamlIssueCache);
+
+  const starlarkIssueCache = cacheIssues(
+    build.parsingResult.issues.filter(issue => issue.path.endsWith('.cirrus.star')),
+  );
+  const starlarkTable = generateTable(build.parsingResult.rawStarlarkConfig, starlarkIssueCache);
+
+  let summaryText: string;
+  let yamlTitle = (
+    <Typography variant="subtitle1">
+      <p>YAML configuration</p>
+    </Typography>
+  );
+  let starlarkTitle = (
+    <Typography variant="subtitle1" className={classes.topPadded}>
+      <p>Starlark configuration</p>
+    </Typography>
+  );
+
+  if (yamlIssueCache.size !== 0 && starlarkIssueCache.size !== 0) {
+    summaryText = 'Failed to parse both YAML and Starlark configurations';
+  } else if (yamlIssueCache.size !== 0) {
+    summaryText = 'Failed to parse YAML configuration';
```

Comment:
Can't issues have warnings only?

Reply:
Good catch! Fixed in https://github.com/cirruslabs/cirrus-ci-web/pull/389/commits/581b1942110a7b1ff0f3d1b9c65a5006d022b29e.